graphql_compiler/schema_transformation/merge_schemas.py

```diff
@@ -48,79 +69,33 @@ def merge_schemas(schema_id_to_ast):
-
-        # Merge current_ast into merged_schema_ast.
-        # Concatenate new scalars, new directives, and type definitions other than the query
-        # type to definitions list.
-        # Raise errors for conflicting scalars, directives, or types.
-        new_definitions = current_ast.definitions  # List[Node]
-        new_query_type_fields = None  # List[FieldDefinition]
-
-        for new_definition in new_definitions:
-            if isinstance(new_definition, ast_types.SchemaDefinition):
-                continue
-            elif (
-                isinstance(new_definition, ast_types.ObjectTypeDefinition) and
-                new_definition.name.value == current_query_type
-            ):  # query type definition
-                new_query_type_fields = new_definition.fields  # List[FieldDefinition]
-            elif isinstance(new_definition, ast_types.DirectiveDefinition):
-                _process_directive_definition(
-                    new_definition, directives, merged_schema_ast
-                )
-            elif isinstance(new_definition, ast_types.ScalarTypeDefinition):
-                _process_scalar_definition(
-                    new_definition, scalars, name_to_schema_id, merged_schema_ast
-                )
-            elif isinstance(new_definition, (
-                ast_types.EnumTypeDefinition,
-                ast_types.InterfaceTypeDefinition,
-                ast_types.ObjectTypeDefinition,
-                ast_types.UnionTypeDefinition,
-            )):
-                _process_generic_type_definition(
-                    new_definition, current_schema_id, scalars, name_to_schema_id, merged_schema_ast
-                )
-            else:  # All definition types should've been covered
-                raise AssertionError(
-                    u'Missed definition type: "{}"'.format(type(new_definition).__name__)
-                )
+        current_ast = deepcopy(current_ast)
+        _accumulate_types(merged_schema_ast, type_name_to_schema_id, scalars, directives,
+                          current_schema_id, current_ast)
 
-        # Concatenate all query type fields.
-        # Since query_type was taken from the schema built from the input AST, the query type
-        # should never be not found.
-        if new_query_type_fields is None:
-            raise AssertionError(u'Query type "{}" field definitions unexpected not '
-                                 u'found.'.format(current_query_type))
-        # Note that as field names and type names have been confirmed to match up, and types
-        # were merged without name conflicts, query type fields can also be safely merged.
-        query_type_index = 1  # Query type is the second entry in the list of definitions
-        merged_schema_ast.definitions[query_type_index].fields.extend(new_query_type_fields)
+    _add_cross_schema_edges(merged_schema_ast, type_name_to_schema_id, scalars,
```

Comment:
Are we missing the [type equivalence hints](https://github.com/kensho-technologies/graphql-compiler/pull/429/files#diff-3931d7bec621ab228f4814115c364a90R330) argument?

Reply:
never mind-- was viewing an out-of-date file